src/pybind/mgr/dashboard/frontend/src/app/shared/directives/copy2clipboard-button.directive.ts

```diff
@@ -0,0 +1,46 @@
+import { Directive, ElementRef, HostListener, Input, OnInit, Renderer2 } from '@angular/core';
+
+import { ToastsManager } from 'ng2-toastr';
+
+@Directive({
+  selector: '[cdCopy2ClipboardButton]'
+})
+export class Copy2ClipboardButtonDirective implements OnInit {
```

Comment:
Adding a reusable component on shared module deserves a separate commit.

This helps the review, and also future maintenance.

(something to take into account in the future...)